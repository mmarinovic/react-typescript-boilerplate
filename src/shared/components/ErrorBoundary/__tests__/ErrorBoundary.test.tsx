import React from 'react';
import { ErrorBoundary } from '..';
import { render } from '@testing-library/react';

describe('ErrorBoundary', () => {
  it('should match snapshot', () => {
    const { container } = render(<ErrorBoundary>"Hi"</ErrorBoundary>);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('catches error and displays fallback', () => {
    const ThrowError = () => {
      throw new Error('Here is it');
    };

    const { container } = render(
      <ErrorBoundary>
        <ThrowError />
      </ErrorBoundary>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
