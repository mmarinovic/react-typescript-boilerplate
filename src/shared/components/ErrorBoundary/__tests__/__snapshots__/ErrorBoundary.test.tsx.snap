// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary catches error and displays fallback 1`] = `
<h1>
  Something went wrong.
</h1>
`;

exports[`ErrorBoundary should match snapshot 1`] = `"Hi"`;
